### PR TITLE
CB-4502 Consolidate DistroX/SDX/FreeIPA/RDS tagging

### PR DESCRIPTION
**Pull request by doktoric**

### CODE REVIEW

File: template-manager-tag/src/test/java/com/sequenceiq/cloudbreak/DefaultCostTaggingServiceAccountTagValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Mock
+    private CentralTagUpdater centralTagUpdater;
+
+    @InjectMocks
+    private DefaultCostTaggingService underTest;
+
+    @Test
+    void getWithResourceTagValidationWhenUserDefinedResourceTagsIsNull() throws AccountTagValidationFailed {
+        underTest.prepareDefaultTags(tagRequest(null));
```
Comment (topolyai5) [inline template-manager-tag/src/test/java/com/sequenceiq/cloudbreak/DefaultCostTaggingServiceAccountTagValidationTest.java]: hm, copy/paste pattern for test name and tested method?

File: template-manager-tag/src/test/java/com/sequenceiq/cloudbreak/DefaultCostTaggingServiceAccountTagValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+public class DefaultCostTaggingServiceAccountTagValidationTest {
+
+    @Mock
+    private CentralTagUpdater centralTagUpdater;
+
+    @InjectMocks
+    private DefaultCostTaggingService underTest;
+
+    @Test
+    void getWithResourceTagValidationWhenUserDefinedResourceTagsIsNull() throws AccountTagValidationFailed {
```
Comment (topolyai5) [inline template-manager-tag/src/test/java/com/sequenceiq/cloudbreak/DefaultCostTaggingServiceAccountTagValidationTest.java]: what are you testing here?

File: template-manager-tag/src/test/java/com/sequenceiq/cloudbreak/DefaultCostTaggingServiceAccountTagValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        underTest.prepareDefaultTags(tagRequest(null));
+    }
+
+    @Test
+    void getWithResourceTagValidationWhenUserDefinedResourceTagsIsEmpty() throws AccountTagValidationFailed {
```
Comment (topolyai5) [inline template-manager-tag/src/test/java/com/sequenceiq/cloudbreak/DefaultCostTaggingServiceAccountTagValidationTest.java]: what are you testing here?

File: template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/TagTemplateProcessor.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TagTemplateProcessor.class);
+
+    private final Handlebars handlebars = HandlebarUtils.handlebars();
+
+    @Inject
+    private Clock clock;
+
+    public String process(String sourceTemplate, TagPreparationObject model) throws IOException {
+        long started = System.currentTimeMillis();
```
[the remark this answers is not in the document]
Comment (doktoric) [reply template-manager-tag/src/main/java/com/sequenceiq/cloudbreak/tag/TagTemplateProcessor.java]: I think this is different here

File: template-manager-tag/build.gradle [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    testCompile group: 'org.springframework.boot',  name: 'spring-boot-starter',            version: springBootVersion
+    testCompile group: 'org.springframework.boot',  name: 'spring-boot-starter-test',       version: springBootVersion
+    testCompile group: 'org.junit.jupiter',         name: 'junit-jupiter-api',              version: junitJupiterVersion
+    testCompile 'org.hamcrest:java-hamcrest:2.0.0.0'
+    testCompile 'net.javacrumbs.json-unit:json-unit:1.28.2'
+
+    testImplementation group: "org.mockito",               name: "mockito-core",                             version: mockitoVersion
+    testImplementation group: "org.mockito",               name: "mockito-junit-jupiter",                    version: mockitoVersion
+    testImplementation group: 'org.junit.jupiter',         name: 'junit-jupiter-params',                     version: junitJupiterVersion
```
[the remark this answers is not in the document]
Comment (doktoric) [reply template-manager-tag/build.gradle]: I think that PR will not solve all the problem just specifically in one modul

[the remark this answers is not in the document]
Comment (topolyai5) [reply template-manager-tag/build.gradle]: No, Laci wants to clean from all module. In the new module we should follow the clean way.

[the remark this answers is not in the document]
Comment (doktoric) [reply template-manager-tag/build.gradle]: done

**Pull request merged**